pkg/controller/volume/expand/sync_volume_resize.go

```diff
@@ -61,10 +61,22 @@ func (rc *syncResize) Sync() {
 			glog.V(10).Infof("Operation for PVC %v is already pending", pvcWithResizeRequest.UniquePvcKey())
 			continue
 		}
-		growFuncError := rc.opsExecutor.GrowPvc(pvcWithResizeRequest, rc.resizeMap)
+		growFuncError := rc.opsExecutor.ExpandVolume(pvcWithResizeRequest, rc.resizeMap)
 		if growFuncError != nil {
 			glog.Errorf("Error growing pvc with %v", growFuncError)
 		}
 		glog.Infof("Resizing PVC %s", pvcWithResizeRequest.CurrentSize)
 	}
+
+	// For PVCs whose API objects updates failed the first time, try again
+	for _, pvcWithUpdateNeeded := range rc.resizeMap.GetPvcsWithUpdateNeeded() {
```

Comment:
So rather than doing this - I think it would be better to not mark PVCs for which updating API object failed as done. The idea would be - if the controller still sees these PVCs in need of resize, then controller will go ahead and attempt resize again. In worst case, this would mean that - we will attempt to resize the volume to a size it has already been resized, in which case the plugin's resize function would return success and we can attempt to update API objects again. I can make this change myself but just wanted to know your thoughts.